src/Passes/Source/RuleTransformationPass/RulePass.cpp

```diff
@@ -0,0 +1,863 @@
+                            if (blocks_queued.find(bb) == blocks_queued.end())
+                            {
+                                queue.push_back(bb);
+                                blocks_queued.insert(bb);
+                            }
+                        }
+                    }
+                    continue;
+                }
+
+                // Follow the branches of
+                auto* switch_instr = llvm::dyn_cast<llvm::SwitchInst>(&instr);
+                if (switch_instr != nullptr)
+                {
+                    for (uint64_t i = 0; i < switch_instr->getNumSuccessors(); ++i)
+                    {
+                        auto bb = switch_instr->getSuccessor(static_cast<uint32_t>(i));
+                        // Ensuring that we are not scheduling the same block twice
+                        if (blocks_queued.find(bb) == blocks_queued.end())
+                        {
+                            queue.push_back(bb);
+                            blocks_queued.insert(bb);
+                        }
+                    }
+                    continue;
+                }
+
+                // Checking if this is an invoke call
+                auto* invoke_inst = llvm::dyn_cast<llvm::InvokeInst>(&instr);
+                if (invoke_inst != nullptr)
+                {
+                    // Checking that configuration is correct
+                    if (!config_.assumeNoExceptions())
+                    {
+                        // TODO(tfr): Unify error reporting
+                        throw std::runtime_error("Exceptions paths cannot be handled at compile time. Either disable "
+                                                 "transform-execution-path-only or add assumption assume-no-except");
+                    }
+
+                    // Adding the block which is the on the "no except" path
+                    auto bb = invoke_inst->getNormalDest();
+                    if (blocks_queued.find(bb) == blocks_queued.end())
+                    {
+                        queue.push_back(bb);
+                        blocks_queued.insert(bb);
+                    }
+                }
+            }
+        }
+
+        // Deleting constants
+        for (auto& x : schedule_instruction_deletion)
+        {
+            x->eraseFromParent();
+        }
+
+        --depth_;
+
+        return true;
+    }
+
+    bool RuleTransformationPass::isActive(llvm::Value* value) const
+    {
+        return active_pieces_.find(value) != active_pieces_.end();
+    }
+
+    void RuleTransformationPass::runCopyAndExpand(llvm::Module& module, llvm::ModuleAnalysisManager&)
+    {
+        replacements_.clear();
+        // For every instruction in every block, we attempt a match
+        // and replace.
+        for (auto& function : module)
+        {
+            if (function.hasFnAttribute(config_.entryPointAttr()))
+            {
```

Comment:
`function` is a [Standard Library identifier](https://en.cppreference.com/w/cpp/utility/functional/function). Consider renaming. Otherwise we can hit an unexpected behavior at some point.

Reply:
Changed.